src/ChainRouter.php

```diff
@@ -227,8 +231,19 @@ public function generate($name, $parameters = [], $absolute = UrlGeneratorInterf
                 continue;
             }
 
+            // if $router does not announce it is capable of handling
+            // non-string routes and the ROUTE_OBJECT is set in the parameters array, continue
+            if (array_key_exists(RouteObjectInterface::ROUTE_OBJECT, $parameters) && is_object($parameters[RouteObjectInterface::ROUTE_OBJECT]) && !$router instanceof VersatileGeneratorInterface) {
```

Comment:
i would not do this check. we can't know how the router wants to work, and if the route name is our "magic" route, the router will simply not find a route to generate. (see my other comment below)

Reply:
Good point, I will revert this!